### DISCUSSION TITLE
Having a system wide installation

### DISCUSSION THREAD

**Discussion by djibb** [mu-editor/mu#2095] [Ideas & New Features]
on W10, with MSI, mu-deitor installing in a wenv in user Appdata directory.

I'm unsing mu-editor on educationnal purpose on a college : 
1296 venv to create, 200 Mo each... -> KABOUM on server.

So, is there a way to install it on c:\programfiles ?

**Comment by tmontes** [comment]
@djibb,

...yes, it is possible. Mu is packaged with [`pup`](https://pup.readthedocs.io/) and per the current capabilities, `pup`-generated MSI files can be installed system-wide by:

1. Bringing up a CLI with elevated privileges.
2. Running `msiexec /i <msi-file>" MSIINSTALLPERUSER=""` there.

Let us know how it goes for you!

**Reply by zabero** [reply]
Hi @tmontes 

We have same issue when we are trying to install it for all students.
I was trying to replicate this with command:
msiexec /i "MuEditor-win64-1.2.0.msi" MSIINSTALLPERUSER="" TARGETDIR="C:\Program Files" /qn 

Sadly without success. 
Can you be so kind and advice how to install it for all users to target directory? 

Kind regards
Zabero

**Reply by tmontes** [reply]
Not sure about the details, off of my head, @zabero, but two thoughts come to my mind:

1. What happens when you install with that command?

2. Can you try the exact command I mentioned, please (that's the one I tested before, and it worked... maybe changing the target install dir leads to the failure?)

Please, share your progress and/or more details. :)

**Comment by zabero** [comment]
Hi, 
1. It was looking he installs it, but I could not find it on the drive anywhere. 
2.  It worked out, but I would prefer to install it to the directory I specified.

**Reply by PrescotAdmin** [reply]
use **allusers="1"** this will install it to program files (x86)

UPDATE - Installs it no problem but crashes on running have a bug report submitted will update my reply once I know anything.

**Comment by PrescotAdmin** [comment]
Hi i'm trying to roll out MuEditor with SCCM to our Computer Lab's but it will only install for user. so ends up in the users appdata\program files. This is no good i need it to install to c:\Program files\MuEditor on each machine. so that i can then push out the shortcut for the users.

I've tried the above ram it on a clean test VM whilst logged in as my admin account and every time it installs in the appdata. This is so annoying. 

Anyone got this to work on a per machine basis installing to c:\Programs ?

**Reply by PrescotAdmin** [reply]
OK update got this to work need to use msiexec /i "MuEditor-win64-1.2.0.msi" ALLUSERS="1" /qn /norestart that works from cmd line now just got to test it through SCCM. This installs it in c:\program Files(x86) which is good enough for me.

**Reply by PrescotAdmin** [reply]
UPDATE - Installs it no problem but crashes on running have a bug report submitted will update my reply once I know anything.
